test/async-hooks/test-callback-error.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const spawnSync = require('child_process').spawnSync;
+const async_hooks = require('async_hooks');
+const initHooks = require('./init-hooks');
+
+switch (process.argv[2]) {
+  case 'test_init_callback':
+    initHooks({
+      oninit: common.mustCall(() => { throw new Error('test_init_callback'); })
```

Comment:
😕  But IMO the actual intention here is just to ensure that the `common.mustCall()` will run only when the `test_init_callback` case is hit?  Every switch case will be hit in different process because of `spawnSync()`s below.